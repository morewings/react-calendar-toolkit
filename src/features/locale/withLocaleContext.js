import React, {useContext} from 'react';
import PropTypes from 'prop-types';
import defaultLocale from 'date-fns/locale/en-US';

const LocaleContext = React.createContext({});

export const useLocaleContext = () => useContext(LocaleContext);

export const Provider = ({children, dateFnsLocale}) => (
  <LocaleContext.Provider value={dateFnsLocale}>
    {children}
  </LocaleContext.Provider>
);

Provider.propTypes = {
  dateFnsLocale: PropTypes.shape({}),
  children: PropTypes.element.isRequired,
};
Provider.defaultProps = {
  dateFnsLocale: defaultLocale,
};

export default WrappedComponent => ({dateFnsLocale, ...restProps}) => (
  <Provider dateFnsLocale={dateFnsLocale}>
    <WrappedComponent {...restProps} />
  </Provider>
);
