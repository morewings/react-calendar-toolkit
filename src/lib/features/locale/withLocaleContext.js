import React, {useContext} from 'react';
import PropTypes from 'prop-types';
import defaultLocale from 'date-fns/locale/en-US';

const LocaleContext = React.createContext({});

export const useLocaleContext = () => useContext(LocaleContext);

export const Provider = ({children, dateFnsLocale}) => (
  <LocaleContext.Provider value={dateFnsLocale}>
    {children}
  </LocaleContext.Provider>
);

Provider.propTypes = {
  dateFnsLocale: PropTypes.shape({}),
  children: PropTypes.element.isRequired,
};
Provider.defaultProps = {
  dateFnsLocale: defaultLocale,
};

const withLocaleContext = WrappedComponent => ({
  // eslint-disable-next-line react/prop-types
  dateFnsLocale,
  ...restProps
}) => (
  <Provider dateFnsLocale={dateFnsLocale}>
    <WrappedComponent {...restProps} />
  </Provider>
);

export default withLocaleContext;
