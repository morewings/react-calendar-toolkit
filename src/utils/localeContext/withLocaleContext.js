/*eslint-disable*/
import React, {useContext} from 'react';
import PropTypes from 'prop-types';
import defaultLocale from 'date-fns/locale/en-US';

export const LocaleContext = React.createContext({});

export const useLocaleContext = () => useContext(LocaleContext);

export const Provider = ({children, value}) => (
  <LocaleContext.Provider value={value}>
    {children}
  </LocaleContext.Provider>
);

Provider.propTypes = {
  value: PropTypes.shape({}),
};
Provider.defaultProps = {
  value: defaultLocale,
};

export default WrappedComponent => ({dateFnsLocale, ...restProps}) => (
  <Provider value={dateFnsLocale}>
    <WrappedComponent {...restProps} />
  </Provider>
);