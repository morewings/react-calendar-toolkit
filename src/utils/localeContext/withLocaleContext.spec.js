import React from 'react';
import {render} from '@testing-library/react';
import {renderHook} from '@testing-library/react-hooks';
import defaultLocale from 'date-fns/locale/en-US';
import withLocaleContext, {
  useLocaleContext,
  Provider,
} from './withLocaleContext';

const Mock = jest.fn(() => 'Mock');

describe('withLocaleContext', () => {
  beforeEach(() => {
    Mock.mockClear();
  });

  it('returns valid React Component', () => {
    const Wrapped = withLocaleContext(Mock);
    expect(Wrapped).toBeInstanceOf(Function);
    const {container} = render(<Wrapped />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('passes props to provided Component', () => {
    const Wrapped = withLocaleContext(Mock);
    const props = {
      hello: 'world',
    };
    const customLocale = {
      foo: 'bar',
    };
    render(<Wrapped {...props} dateFnsLocale={customLocale} />);
    expect(Mock.mock.calls[0][0]).toEqual(props);
  });
});

describe('useLocaleContext', () => {
  it('returns en-US locale by default', () => {
    const {result} = renderHook(() => useLocaleContext(), {
      wrapper: ({children}) => <Provider>{children}</Provider>,
    });

    expect(result.current).toBe(defaultLocale);
  });

  it('returns custom locale when set', () => {
    const customLocale = {
      foo: 'bar',
    };

    const {result} = renderHook(() => useLocaleContext(), {
      wrapper: ({children}) => (
        <Provider value={customLocale}>{children}</Provider>
      ),
    });

    expect(result.current).toBe(customLocale);
  });
});
