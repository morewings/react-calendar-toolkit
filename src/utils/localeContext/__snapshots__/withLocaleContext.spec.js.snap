// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLocaleContext returns valid React Component 1`] = `Mock`;
